janusgraph-core/src/main/java/org/janusgraph/core/attribute/Geoshape.java

```diff
@@ -547,6 +548,16 @@ public Geoshape read(ScanBuffer buffer) {
             try {
                 return GeoshapeBinarySerializer.read(inputStream);
             } catch (IOException e) {
+                if (buffer instanceof ReadArrayBuffer && l == 1) {
+                    // retry using legacy point deserialization
+                    ((ReadArrayBuffer) buffer).movePositionTo(0);
+                    VariableLong.readPositive(buffer);
+                    try {
+                        return point(buffer.getFloat(), buffer.getFloat());
```

Comment:
Can you add a new point() method that takes one argument instead of two? seems like you are calling the same method twice here.